### PR TITLE
Adds the message server key to the Chief Engineer's memories.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
Like the captain has innate knowledge of the golden safe code, the Chief Engineer should know the daily key of the message server.

## Why It's Good For The Game
It's a perk of the job.

## Changelog

:cl:
add: Added the daily (roundstart) message server key to the Chief Engineer's memories.
/:cl:

### CODE REVIEW

File: code/game/machinery/telecomms/machines/message_server.dm [hunk 1]
```diff
@@ -172,9 +164,23 @@
...
 	calibrating = 0
 
+GLOBAL_VAR(preset_station_message_server_key)
+
+/obj/machinery/telecomms/message_server/preset/Initialize(mapload)
+	. = ..()
+	// Just in case there are multiple preset messageservers somehow once the CE arrives,
+	// we want those on the station to share the same preset default decrypt key shown in his memories.
+	var/is_on_station = is_station_level(z)
+	if(is_station_level(z) && GLOB.preset_station_message_server_key)
```
Comment (distributivgesetz) [inline code/game/machinery/telecomms/machines/message_server.dm:176]:
```suggestion
	if(is_on_station && GLOB.preset_station_message_server_key)
```

**Comment by tralezab** [comment]
I always love these because it lets changelings slurp up interesting information.

**Comment by mikederkan** [comment]
Great change

**Comment by Ghommie** [comment]
> I always love these because it lets changelings slurp up interesting information.

Also paradox clones.

File: code/game/machinery/telecomms/machines/message_server.dm [hunk 2]
```diff
@@ -127,11 +124,6 @@
  * it in this proc, you have to do so yourself.
  */
 /obj/machinery/telecomms/message_server/proc/generate_key()
```
Comment (san7890) [inline code/game/machinery/telecomms/machines/message_server.dm:126]: do subtypes use this proc somehow? i don't get why you stubbed it out while moving all the code it contains elsewhere

**Pull request merged**